src/ty.md

```diff
@@ -4,65 +4,130 @@ The `ty` module defines how the Rust compiler represents types
 internally. It also defines the *typing context* (`tcx` or `TyCtxt`),
 which is the central data structure in the compiler.
 
-## The tcx and how it uses lifetimes
+## `ty::Ty`
 
-The `tcx` ("typing context") is the central data structure in the
-compiler. It is the context that you use to perform all manner of
-queries. The struct `TyCtxt` defines a reference to this shared context:
+When we talk about how rustc represents types,  we usually refer to a type called `Ty` . There are quite a few modules and types for `Ty` in the compiler ([Ty documentation](https://doc.rust-lang.org/nightly/nightly-rustc/rustc/ty/index.html)). 
 
-```rust,ignore
-tcx: TyCtxt<'tcx>
-//          ----
-//          |
-//          arena lifetime
-```
+The specific `Ty` we are referring to is `rustc::ty::Ty` (and not `rustc::hir::Ty`). The distinction is important, so we will discuss it first before going into the details of `ty::Ty`.
```

Comment:
we should link to the API docs when mentioning types. e.g.

* https://doc.rust-lang.org/nightly/nightly-rustc/rustc/ty/type.Ty.html
* https://doc.rust-lang.org/nightly/nightly-rustc/rustc/hir/struct.Ty.html

this can also help us detect bitrot

Reply:
Fixed.